digits/static/js/pretrained_models/LayerVisualizations.js

```diff
@@ -0,0 +1,356 @@
+    self.outer.style("padding", "30px");
+
+    var panel   = self.outer.append("div").attr("class", "panel panel-default")
+
+    var heading = panel.append("div").attr("class", "panel-heading")
+      .append("div").attr("class","row").style("padding","0px 10px");
+
+    self.headingCenter = heading.append("div").attr("class", "text-center col-xs-offset-1 col-xs-10");
+    self.headingCenter.append("div").html(parent.layer.label + " (" + parent.layer.stats.shape + ")");
+    self.headingRight  = heading.append("div").attr("class", "col-xs-1 text-right");
+
+    var panelBody = panel.append("div").attr("class", "panel-content");
+    self.body = panelBody.append("div").attr("class", "outputs");
+    self.body.append("div")
+      .attr("class", "alert alert-"+status)
+      .style("margin","15px")
+      .html(msg);
+    self.nav = panelBody.append("div").attr("class", "panel-nav");
+
+    self.drawCloseButton();
+  };
+
+  // Event Listeners:
+  d3.select("body").on("keydown", function(){
+    if (d3.event.keyCode == 27) self.remove();
+  });
+};
+
+LayerVisualizations.Overlay = function(selector,props){
+  var self   = this;
+  var props  = _.isUndefined(props) ? {} : props;
+  var parent = !_.isUndefined(props.parent) ? props.parent : props;
+
+  self.container = d3.select(selector);
+  self.inner = null;
+
+  self.render = function(){
+    self.remove();
+    self.container.style("position","relative");
+    self.inner = self.container.append("div")
+      .attr("class", "text-center component-outer loading-overlay");
+
+    self.inner.style("background", "rgba(0,0,0,0.5)");
+
+    var spinner = self.inner.append("span")
+      .attr("class", "glyphicon glyphicon-refresh glyphicon-spin");
+
+    spinner.style({top: "50%", "margin-top": "-20px", color: "white", "font-size": "40px"})
+
+    return self.inner;
+  };
+
+  self.remove = function(){
+    d3.selectAll(".loading-overlay").remove();
+  };
+
+};
+
+LayerVisualizations.Jobs = function(selector,props){
+  var self   = this;
+  var parent = !_.isUndefined(props.parent) ? props.parent : props;
+
+  self.jobs      = props.jobs;
+  self.container = d3.select(selector);
+  self.layers_with_outputs = [];
+
+  self.update = function(){
+    var items = self.container.selectAll("div").data(self.jobs);
+    items.enter()
+      .append("div")
+        .attr("class","btn btn-xs btn-default")
+        .style(LayerVisualizations.Styles.button);
+
+    items.append("div").html(function(d){return d.name});
+    items.append("div").attr("class","subtle").html(function(d){return d.id});
+    items.on("click",self.dispatch);
+
+    items.exit().remove();
+  };
+
+  self.dispatch = function(d){
+    parent.actions.getJob(d.id);
+  }
+
+  self.load = function(json){
+    if (json.framework == "caffe"){
+      generateCaffeTree(json.model_def);
+      self.layers_with_outputs = _.pluck(_.filter(graph.nodes, function(n){return !_.isUndefined(n.num_outputs)}),"name");
```

Comment:
I don't think checking for `num_output` is necessarily going to work for all layers.